File: bubblesort/php/BubbleSort.php
<?php 

function bubble(&$arr)
{
  $isSorted = false;
  $lastSorted = sizeof($arr) - 1;

  while (!$isSorted) {
    $isSorted = true;

    for ($i = 0; $i < $lastSorted; $i++) {
      if ($arr[$i] > $arr[$i + 1]) {
        swap($arr, $i, $i + 1);
        $isSorted = false;
      }
    }
    $lastSorted--;
  }
}

function swap(&$arr, $i, $j)
{
  $temp = $arr[$i];
  $arr[$i] = $arr[$j];
  $arr[$j] = $temp;
}

function printArr(&$arr)
{
  echo "[";
  for ($i = 0; $i < sizeof($arr); $i++) {
    echo $arr[$i];
    if ($i != sizeof($arr) - 1)
      echo ", ";
  }
  echo "]\n";
}
$nums = [4, 3, 5, 9, 1, 2, 7, 8];

printArr($nums);
bubble($nums);
printArr($nums);

File: bubblesort/dart/BubbleSort.dart
void bubble(List<int> arr) {
  bool isSorted = false;
  int lastSorted = arr.length - 1;

  while (!isSorted) {
    isSorted = true;
    for (int i = 0; i < lastSorted; i++) {
      if (arr[i] > arr[i + 1]) {
        swap(arr, i, i + 1);
        isSorted = false;
      }
    }
    lastSorted--;
  }
}

void swap(List<int> arr, int i, int j) {
  int temp = arr[i];
  arr[i] = arr[j];
  arr[j] = temp;
}

main() {
  List<int> nums = [3, 6, 8, 9, 1, 4, 2, 5, 7];
  print(nums);
  bubble(nums);
  print(nums);
}
